### PR TITLE
Fixed #31268 -- Updated Algerian Arabic (ar_DZ) locale formats.

### PR DESCRIPTION
**Pull request by infosrabah**

### CODE REVIEW

File: django/conf/locale/ar_DZ/formats.py [hunk 1]
```diff
@@ -2,20 +2,30 @@
...
 # The *_INPUT_FORMATS strings use the Python strftime format syntax,
 # see https://docs.python.org/library/datetime.html#strftime-strptime-behavior
-# DATE_INPUT_FORMATS =
-# TIME_INPUT_FORMATS =
-# DATETIME_INPUT_FORMATS =
+DATE_INPUT_FORMATS = [
+    '%Y/%m/%d', '%y/%m/%d',  # '2006/10/25', '06/10/25'
+]
+TIME_INPUT_FORMATS = [
+    '%H:%M',        # '14:30
```
Comment (felixxm) [inline django/conf/locale/ar_DZ/formats.py]: Please chop trailing whitespaces.

File: django/conf/locale/ar_DZ/formats.py [hunk 2]
```diff
@@ -2,20 +2,30 @@
...
-TIME_FORMAT = 'g:i A'
-# DATETIME_FORMAT =
+DATE_FORMAT = 'j F Y'  
+TIME_FORMAT = 'H:i'
+DATETIME_FORMAT = 'j F Y H:i'
 YEAR_MONTH_FORMAT = 'F Y'
 MONTH_DAY_FORMAT = 'j F'
-SHORT_DATE_FORMAT = 'd‏/m‏/Y'
-# SHORT_DATETIME_FORMAT =
+SHORT_DATE_FORMAT = 'Y/m/d'
```
Comment (felixxm) [inline django/conf/locale/ar_DZ/formats.py]: Are you sure? Should this not be consistent with `SHORT_DATETIME_FORMAT`, i.e. `SHORT_DATE_FORMAT = 'j N Y'`.

Comment (infosrabah) [reply django/conf/locale/ar_DZ/formats.py]: Hi Felix,
You right for the SHORT_DATE_FORMAT

To summerize you, maybe you can help Felix, 

My application is a multilanguage app, mean needs a translations and formats, that's the reason why I needed a ar-DZ, but now I want to send that format used in django to a bootstrap datepicker, means that in our input case : DATE_INPUT_FORMATS : '%Y/%m/%d' I want to use it to convert it so that bootstrap datetime can use that format, like : 'yyyy/mm/dd', I know how to convert and use a snippet for that, but I didn't know where I can get that django date_input format that has been used to generate the field  'Y/m/d'. but the input in form stick to "d/m/Y' ??


Do you know how to get the current format used/applied to a date_field ? I wan't pass that to a bootstrap datepicker.

Here's a screenshot of a Update view well rendering date format in the ar-DZ :
<a>https://ibb.co/XCvC7rS</a>
and when you try to put a date into it using datebootstrap, wich is not well formated :
PS : I didn't define a date-format for bootstrap datepicker, I let it use the django default format country.
<a>https://ibb.co/7xfNY80</a>

Hope to hear from you soon.

Comment (felixxm) [reply django/conf/locale/ar_DZ/formats.py]: @infosrabah PR is not the best place to ask support question. Please use one of [ support channels](https://code.djangoproject.com/wiki/TicketClosingReasons/UseSupportChannels).

**Comment by felixxm** [comment]
@infosrabah Apart from you support question, Should we update formats for Algerian Arabic?

(\cc @claudep).

**Comment by claudep** [comment]
It's very important that the format settings represent the real way date and time is presented in your country in Arabic. It should absolutely have nothing to do with any technical problem you may have with bootstrap or any other tool.

For example, on Wikipedia (https://en.wikipedia.org/wiki/Algeria), the documented date format is dd/mm/yyyy (`d/m/Y`).

**Comment by infosrabah** [comment]
Yues update the format Felix, please.
----

Hi claude,

I agree about what you said,

But we do have 2 major languages in Algeria(French and Arabic), the date that your are refering to is for fr-DZ wich is dd/mm/yyyy (d/m/Y). but I chere we are talking about ar-DZ wich is completly the RTL way (opposite) : yyyy/mm/dd (Y/m/d). I may in the future create a fr-DZ formats.py wich is almost the same as fr-FR.

And beside of that I do not witness any technical problem you mentioned about bootstrap or any other tool. I just asked Felix about the way to get the actual coutry format that has been applied to a date. just to convert that and pass it to front-end tool. So there will be some consistency between the two world (front-end and backend), and that makes your life much easier when it comes about (Validation).

Thank you Claude

**Comment by claudep** [comment]
As I can see, the current format of RTL languages like Arabic, Hebrew and others are still using the `d‏/m‏/Y` syntax in format settings. Maybe something to ask on the django-i18n mailing list?

**Comment by infosrabah** [comment]
Hi Claude,

Here's as examples of some local news paper online website and even a lcoal TV website formats used : (dates are surrounded green)

1 - News paper 1 <a>https://ibb.co/d6Cyshd</a>
2 - News paper 2 (Sport) <a>https://ibb.co/frrd9fP</a>
3 - Local TV <a>https://ibb.co/85Bh53R</a>

Now if most RTL other countries are using or still using 'd‏/m‏/Y ' as you said, my country is not. as you said you have to make it reflect as in real life (that's what I'm trying to do). 

For every date on any ressource (News, Official communications, News papper) that using Arabic as a language, the Date format is 'Y/m/d'.

About the wikipedia that you posted about Algeria and the Date format used 'dd/mm/yyyy' (d/m/Y). This applies for the French language in Algeria (fr-DZ) LTR, not the arabic (ar-DZ) that we are talking about here.

Hope I shed more light on this post and gave you more reasons to convice you that the Date format for Arabic Algeria (ar-DZ) is 'Y/m/d'.



Thank you.

**Comment by claudep** [comment]
That's very precious data, thanks! Now may I ask you to create a ticket and provide the information you just wrote to that ticket? Ticket data is supposed to be kept long term, contrary to pull request comments.

Are you also able to confirm that what you said is only for short dates, and that the "normal" date format is still "day month year" (`j F Y`)?

**Comment by infosrabah** [comment]
Hi claude,

Thank's for the quick answer.

For the nomal date the format is the same as you said. ( j F Y ) I confirm it. I used this format ( j F Y ) and the date rendred well in djkango tempalte for RTL, you can refer to the news paper number 1 in the top of the page surrounded green. so far so good.


Now do you want me to go to "Django's bug tracker and wiki" and create a new ticket ??

If so, what do you want me to say out-there ???, this is not a bug/issue, this is settings country format for a specific language. !!

Hope to hear from you soon.

**Comment by felixxm** [comment]
@infosrabah I'm also confused, TBH :smile: Please create [a new ticket](https://code.djangoproject.com/newticket) and describe what is incorrect/missing in the current Algerian Arabic (`ar_DZ`) locale formats. Should we update only `SHORT_DATE_FORMAT`? or maybe all change from this PR are required.

This is still an issue in Django even if it's only in a country specific format.

**Comment by infosrabah** [comment]
HI Felix, 

All change from this PR are required.

Now for the "SHORT_DATE_FORMAT" is little bit different when it comes about arabic, to be specific there's isn't, meaning we do not have Short version of month like "Jan","Fev"... "Dec", We do have a whole Month names "January" in Arabic, Month can't be can't trunc to get a Short version.

I made a second pull request wich fix everrrrrrything, So yes Update all changes.

I'm gonna create a new ticket sure.

Thank you Felix

**Comment by felixxm** [comment]
@infosrabah Thanks :+1: I changed `SHORT_DATETIME_FORMAT = 'j N Y H:i'` to `'Y/m/d H:i'`, because if abbreviations of months don't exist in Algerian Arabic we shouldn't use `N` in formats. Can you confirm the current form?

**Comment by infosrabah** [comment]
Ticket created at : https://code.djangoproject.com/ticket/31268

thank you

**Comment by infosrabah** [comment]
Yes I confirm it Felix,

I have a lots in mind, so it's kind of hard to follow, LoL.

Thank you

**Comment by felixxm** [comment]
> But What about (DATE_INPUT_FORMATS, TIME_INPUT_FORMATS, DATETIME_INPUT_FORMATS) ????,
> really interesting about inputs format to be reflected to django forms.

I'm not really understand what's the question? Please approve this PR if these changes work for you and if something should be changed you can push changes.
